symengine/cwrapper.cpp

```diff
@@ -568,6 +568,27 @@ void basic_str_free(char *s)
     delete[] s;
 }
 
+int symengine_have_component(const char *c)
+{
+#ifdef HAVE_SYMENGINE_MPFR
+    if (std::strcmp("mpfr", c) == 0)
+        return 1;
+#endif
+#ifdef HAVE_SYMENGINE_MPC
+    if (std::strcmp("mpc", c) == 0)
+        return 1;
+#endif
+#ifdef HAVE_SYMENGINE_FLINT
+    if (std::strcmp("flint", c) == 0)
+        return 1;
+#endif
+#ifdef HAVE_SYMENGINE_ARB
+    if (std::strcmp("arb", c) == 0)
+        return 1;
+#endif
```

Comment:
If you want to add all the components, here is a list, https://github.com/symengine/symengine/blob/master/symengine/symengine_config.h.in#L13-L41